### PR TITLE
Add the code-splitting article

### LINKED ISSUE #1060: content: Route-level code-splitting in Angular
**A one to two sentence description of your post**
What are you planning to teach folks in this post?

**Target publish date:** `<2019-07-16>`
- [ ] Check this box if this is a hard deadline.

**Process**
Make a copy of [this template](https://docs.google.com/document/d/1ydauCufwwavStaKxhIDHuNivVxgQBOdDno4uLjGIjmE/edit)
and fill in the proposal and outline sections. When you're finished, drop a link
to the doc in the 1st step below and check the box. A member of the web.dev team
will assign this ticket to themselves and work as your reviewer to help you
complete the remaining steps.

- [x] 1. I've done my proposal https://docs.google.com/document/d/1DtfQF9wFB01CZjjUCf091eWmaMr8it6b7tnj279ByVQ/edit
- [x] 2. Proposal approved — _reviewer checks this_
- [x] 3. I've added a draft to my proposal doc
- [x] 4. Draft approved — _reviewer checks this_
- [x] 5. I've submitted a pull request #1113
- [x] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by mgechev**
Fixes #1060

Image from [here](https://pixabay.com/photos/puzzle-play-piecing-together-673250/). Content for [route-level code-splitting](https://docs.google.com/document/d/1DtfQF9wFB01CZjjUCf091eWmaMr8it6b7tnj279ByVQ/edit?usp=sharing).

### CODE REVIEW

File: src/site/content/en/angular/route-level-code-splitting/index.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+</div>
+
+In this post we optimized the app's initial load time at the cost of slowing down navigation to pages. In the [next post](/route-preloading) on route preloading we'll see how we can workaround this problem!
+
+## Conclusion
+
+You can shrink the bundle size of your Angular applications by applying route-level code-splitting:
+
+1. Introduce route-level code-splitting in an Angular application.
+1. Use the `ng g module nyan --module app --route nyan` generator to automatically scaffold a dynamically loaded route
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/route-level-code-splitting/index.md]: Can we generalize this for readers who aren't trying to create a nyan cat app? :)

**Comment by mgechev** [comment]
@mfriesenhahn I just addressed your comments. Let me know if I'm missing anything.

**Comment by mfriesenhahn** [comment]
Caught two small issues, so I just took care of them in a commit. Moving to approved!

File: src/site/content/en/angular/angular.11tydata.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ module.exports = {
       {
         title: 'Fast',
         pathItems: [
+          'route-level-code-splitting',
```
Comment (robdodson) [inline src/site/content/en/angular/angular.11tydata.js]: @mgechev should this post go after the one on performance budgets.

I noticed at the end of this post there is a warning which mentions:

```
Warning: Code-splitting can significantly improve an app's initial load time, but it comes at the cost of slowing down subsequent navigation. In the next post on route preloading you'll see how to work around this problem!
```

But with this setup the next post will be performance budgets, not preloading.

**Comment by mgechev** [comment]
@robdodson just pushed a [fix](https://github.com/GoogleChrome/web.dev/pull/1113/commits/1c18b01923b6acada0195182e97fe708fe2aece6). Thanks for pointing this out!

**Pull request merged**